packages/x-charts-pro/src/ChartsToolbarPro/ChartsToolbarPro.tsx

```diff
@@ -1,22 +1,52 @@
-import { Toolbar, ChartsToolbarProps, ToolbarButton } from '@mui/x-charts/Toolbar';
+'use client';
+
 import * as React from 'react';
+import PropTypes from 'prop-types';
+import { Toolbar, ChartsToolbarProps, ToolbarButton } from '@mui/x-charts/Toolbar';
 import { useChartContext, useSelector, useChartsSlots } from '@mui/x-charts/internals';
 import { useChartsLocalization } from '@mui/x-charts/hooks';
+import useId from '@mui/utils/useId';
+import { ChartsToolbarDivider } from './internals/ChartsToolbarDivider';
+import { ChartsMenu } from './internals/ChartsMenu';
 import { selectorChartZoomIsEnabled } from '../internals/plugins/useChartProZoom';
 import { ChartsToolbarZoomInTrigger } from './ChartsToolbarZoomInTrigger';
 import { ChartsToolbarZoomOutTrigger } from './ChartsToolbarZoomOutTrigger';
 import { ChartsSlotsPro } from '../internals/material';
+import {
+  ChartsToolbarPrintExportOptions,
+  ChartsToolbarPrintExportTrigger,
+} from './ChartsToolbarPrintExportTrigger';
+import {
+  ChartsToolbarImageExportOptions,
+  ChartsToolbarImageExportTrigger,
+} from './ChartsToolbarImageExportTrigger';
+
+export interface ChartsToolbarProProps extends ChartsToolbarProps {
+  printOptions?: ChartsToolbarPrintExportOptions;
+  imageExportOptions?: ChartsToolbarImageExportOptions[];
+}
 
-export interface ChartsToolbarProProps extends ChartsToolbarProps {}
+const DEFAULT_IMAGE_EXPORT_OPTIONS: ChartsToolbarImageExportOptions[] = [{ type: 'image/png' }];
 
 /**
  * The chart toolbar component for the pro package.
  */
-export function ChartsToolbarPro(props: ChartsToolbarProProps) {
+function ChartsToolbarPro({
+  printOptions,
+  imageExportOptions: rawImageExportOptions,
+  ...other
+}: ChartsToolbarProProps) {
   const { slots, slotProps } = useChartsSlots<ChartsSlotsPro>();
   const { store } = useChartContext();
   const { localeText } = useChartsLocalization();
+  const [exportMenuOpen, setExportMenuOpen] = React.useState(false);
+  const exportMenuTriggerRef = React.useRef<HTMLButtonElement>(null);
+  const exportMenuId = useId();
+  const exportMenuTriggerId = useId();
   const isZoomEnabled = useSelector(store, selectorChartZoomIsEnabled);
+  const imageExportOptionList = rawImageExportOptions ?? DEFAULT_IMAGE_EXPORT_OPTIONS;
+  const showExportMenu =
+    printOptions?.disableToolbarButton !== true || imageExportOptionList.length > 0;
```

Comment:
Is their a particular edge case I'm missing?
```suggestion
    !printOptions?.disableToolbarButton || imageExportOptionList.length > 0;
```